examples/stm32wl/src/bin/uart_async_task.rs

```diff
@@ -0,0 +1,89 @@
+#![no_std]
+#![no_main]
+#![feature(type_alias_impl_trait)]
+
+use defmt::*;
+use embassy_executor::Spawner;
+use embassy_stm32::gpio::{AnyPin, Level, Output, Pin, Speed};
+use embassy_stm32::interrupt;
+use embassy_stm32::usart::{Config, Uart};
+use embassy_time::{Duration, Timer};
+use {defmt_rtt as _, panic_probe as _};
+
+/*
+Pass Incoming data from LPUART1 to USART1
+Example is written for the LoRa-E5 mini v1.0
+*/
+#[embassy_executor::main]
+async fn main(spawner: Spawner) {
+    let mut config = embassy_stm32::Config::default();
+    config.rcc.mux = embassy_stm32::rcc::ClockSrc::HSE32;
+    let p = embassy_stm32::init(config);
+
+    defmt::info!("Starting system");
+
+    let mut config1 = Config::default();
+    config1.baudrate = 9600;
+
+    let mut config2 = Config::default();
+    config2.baudrate = 9600;
+
+    //RX1/TX1 (LPUART) on LoRa-E5 mini v1.0
+    let irq = interrupt::take!(LPUART1);
+    let mut usart2 = Uart::new(p.LPUART1, p.PC0, p.PC1, irq, p.DMA1_CH5, p.DMA1_CH6, config2);
+
+    //RX/TX connected to USB/UART Bridge on LoRa-E5 mini v1.0
+    let irq = interrupt::take!(USART1);
+    let mut usart1 = Uart::new(p.USART1, p.PB7, p.PB6, irq, p.DMA1_CH3, p.DMA1_CH4, config1);
+
+    unwrap!(usart1.write(b"Hello Embassy World!\r\n").await);
+    unwrap!(usart2.write(b"Hello Embassy World!\r\n").await);
+
+    //PB5 OnBoard-LED on LoRa-E5 mini v1.0
+    unwrap!(spawner.spawn(uart_passthrough(usart1, usart2, p.PB5.degrade())));
+
+    loop {
+        Timer::after(Duration::from_secs(10)).await;
```

Comment:
Rather than having an idle loop here, consider copying the loop from the passthrough task.